### PR TITLE
Add support for external keys and importing custom key material

### PR DESCRIPTION
**Pull request by onematchfox**
This PR adds support for `EXTERNAL` origin keys and the ability of import custom key material. 

One note of the implementation of the key material import process is that I have chosen the easy (lazy?) route of just attaching a single  `ParametersForImport` to an existing key. This means that multiple calls to `GetParametersForImport` will resulting in the older parameters being overwritten. Not sure if this differs from the AWS implementation or not. However this implementation does: 
- offer a simplistic way to test "expired/non-existent" import parameters as one can simply call the get parameters method twice
- simplifies the implementation and reduces database size (over time) as we don't have to worry about cleaning up expired import parameters

Also, I haven't added any tests for this yet. I have looked at the Postman collection and am open to trying to add tests to this collection even if I'm not exactly sure how to generate the required input for `ImportKeyMaterial` (see bash script in README.md for what this looks like). However, when I did import the Postman collection it took me quite a while to get the existing test set running as it relies on a number of variables which seem to be stored in local environment rather than as part of the collection. Might be worth considering other options - native go tests? Am open to ideas/discussion here.

### CODE REVIEW

**Comment by nsmithuk** [comment]
Thanks @onematchfox, I'll take a look at this asap this week.

And I agree the testing needs changing. Postman worked fine as a quick way of getting started, but it's not really conducive to multiple developers. I think perhaps it should be swapped out for something script based. I'd like to keep the tests functional based - i.e. testing from a HTTP perspective as I've built most of this with TDD; writing tests against AWS, then implementing LKMS to match. So native Go sounds a good option. Or Python with Requests perhaps.

File: README.md [hunk 1]
```diff
@@ -223,6 +235,70 @@ awslocal kms generate-data-key \
...
+# If you want to re-import key material then you'll need to save
+# this file and use it for any subsequent calls to Local KMS
+# cp $keyMaterial KeyMaterial-$keyId.txt
+
+echo ""
+echo "Encrypting key material using public key"
+encryptedKeyMaterial=$(mktemp)
+# This command assumes the RSAES_OAEP_SHA_1 wrapping algorithm
+openssl rsautl -encrypt \
+                 -in KeyMaterial.txt \
```
Comment (nsmithuk) [inline README.md:277]:
```suggestion
                 -in $keyMaterial \
```

File: src/seed.go [hunk 2]
```diff
@@ -108,9 +111,13 @@ func seed(path string, database *data.Database) {
 		aesKeys[i].Metadata.CreationDate = time.Now().Unix()
 		aesKeys[i].Metadata.Enabled = true
 		aesKeys[i].Metadata.KeyManager = "CUSTOMER"
-		aesKeys[i].Metadata.KeyState = "Enabled"
```
Comment (nsmithuk) [inline src/seed.go:111]: I think we still want the default `KeyState` to be Enabled here. And then we override it below if it needs setting to Pending.

So `aesKeys[i].Metadata.KeyState = cmk.KeyStateEnabled`

Or is there a reason we shouldn't be setting it?

**Review by nsmithuk — CHANGES_REQUESTED** [review]
Thank you for doing this - it's awesome!

I've added question regarding the default KeyState in the seeding process KeyState.

File: README.md (hunk 1 above)
Comment (onematchfox) [reply README.md]: Oops. Well spotted. Thanks.

File: src/seed.go (hunk 2 above)
Comment (onematchfox) [reply src/seed.go:111]: Nope, you are quite right. It should be set to `KeyStateEnabled`. ~copy~cut-pasta mistake.

File: README.md (hunk 1 above)
Comment (onematchfox) [reply README.md]: FYI: Actually made a few others changes to simplify this script as well as adding support for different wrapping algorithms.

**Pull request merged**